sympy/solvers/ode/ode.py

```diff
@@ -3693,6 +3709,51 @@ def ode_2nd_linear_bessel(eq, func, order, match):
     return Eq(f(x), ((x**(Rational(1-c4,2)))*(C0*besselj(n/d4,a4*x**d4/d4)
            + C1*bessely(n/d4,a4*x**d4/d4))).subs(x, x-b4))
 
+def ode_2nd_nonlinear_autonomous_conserved(eq, func, order, match):
+    r"""
+    Gives solution for the autonomous second order nonlinear
+    differential equation of the form
+
+    .. math :: f''(x) = g(f(x))
+
+    The solution for this differential equation can be computed
+    by multiplying by `f'(x)` and integrating on both sides,
+    converting it into a first order differential equation.
+
+    Examples
+    ========
+
+    >>> from sympy import Function, Symbol, dsolve
+    >>> f = Function('f')
+    >>> x = Symbol('x')
+
+    >>> from sympy import sqrt
+    >>> eq = f(x).diff(x, 2) + sqrt(f(x))
+    >>> dsolve(eq)
+    [Eq(2*f(x)*gamma(2/3)*hyper((1/2, 2/3), (5/3,), 4*exp_polar(2*I*pi)*f(x)**(3/2)/(3*C1))/(3*sqrt(C1)*gamma(5/3)), C2 + x),
+    Eq(2*f(x)*gamma(2/3)*hyper((1/2, 2/3), (5/3,), 4*exp_polar(2*I*pi)*f(x)**(3/2)/(3*C1))/(3*sqrt(C1)*gamma(5/3)), C2 - x)]
+
+    >>> from sympy import exp, log
+    >>> eq = f(x).diff(x, 2) - exp(f(x)) + log(f(x))
+    >>> dsolve(eq)
+    [Eq(Integral(1/sqrt(C1 - 2*u*log(u) + 2*u + 2*exp(u)), (u, f(x))), C2 + x),
+    Eq(Integral(1/sqrt(C1 - 2*u*log(u) + 2*u + 2*exp(u)), (u, f(x))), C2 - x)]
+
+    References
+    ==========
+
+    http://eqworld.ipmnet.ru/en/solutions/ode/ode0301.pdf
+    """
+    x = func.args[0]
+    u = symbols('u')
+    f = func.func
+    g = match['g']
+    g = g.subs(f(x), u)
+    C1, C2 = get_numbered_constants(eq, num=2)
+    inside = -2*Integral(g, (u, f(x))).doit().subs(f(x), u) + C1
+    lhs = Integral(1/sqrt(inside), (u, f(x))).doit()
```

Comment:
This should not use doit. It's better to return the formula in unevaluated integrals. Later on `dsolve` will call `doit` but the user should have the option to not call `doit` using:
```
hint = '2nd_nonlinear_autonomous_conserved_Integral'
```

Reply:
We could do that. I notice that, before returning the solutions, the helper functions simplify and remove redundant solutions. However, equations of this kind usually have an unevaluated integral in the solution and it is taking forever to simplify this, and in some cases fails after 10-15 mins. Can I just skip simplification just for this hint? Or else, the solver would practically be useless.

Reply:
Actually skipping simplification would be bad as some solutions are being simplified. Is there any other solution to this?

Reply:
Perhaps in general if `simplify=False` then we should skip the remove redundant solutions step. Or otherwise perhaps remove redundant solutions should only attempt to remove cases where the solutions are linear in the constants.

Otherwise we can improve the way that solutions are simplified if it is being slow.

Reply:
Setting `simplify=False` is not always good as some equations which can be simplified don't get simplified.
I noticed that the simplification is only slow if there is an integral that can't be evaluated. Is there any way to recursively check if an expression contains an `Integral` object? Then, we can check if the hint is `2nd_nonlinear_autonomous_conserved` and if the solution has an Integral, return it as it is.

Reply:
> Otherwise we can improve the way that solutions are simplified if it is being slow.

The call stack looks too complicated. I'm not sure if this would work

Reply:
Yes, but for now the user can try `dsolve(..., simplify=False)` if `dsolve` is slow. We can test some examples with `simplify=False`.

I'm sure that the simplification in `dsolve` can be improved.

Reply:
Ok